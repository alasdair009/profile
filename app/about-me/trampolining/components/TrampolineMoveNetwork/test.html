<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
