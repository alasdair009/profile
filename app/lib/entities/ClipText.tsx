"use client";

import styled from "styled-components";
import {rem} from "polished";

export const ClipText = styled.span`
  align-items: center;
  //background: black;
  display: flex;
  font-size: ${rem(56)};
  height: 100%;
  justify-content: center;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  -webkit-text-fill-color: transparent;
  text-fill-color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  background: black;
  color: white;
`;