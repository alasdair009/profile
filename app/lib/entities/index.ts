"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
