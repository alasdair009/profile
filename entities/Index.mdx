import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
