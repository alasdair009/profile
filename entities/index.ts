export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
