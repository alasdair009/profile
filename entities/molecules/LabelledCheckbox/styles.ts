"use client";

import styled from "styled-components";

export const Root = styled.div`
  display: inline-flex;
  align-items: center;
`;
