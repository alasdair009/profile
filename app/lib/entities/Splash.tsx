import {HTMLAttributes} from "react";
import styled from "styled-components";
import {globalContentMaxWidth} from "@/entities/design-tokens/dimensions";
import {rem} from "polished";
import {Header, Heading, Paragraph} from "@/entities";

type SplashProps = {
} & HTMLAttributes<HTMLDivElement>;

const Root = styled.section`
  height: calc(100vh - ${Header.height});
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  position: relative;
  width: 100%;
`;

const SplashCopy = styled.div`
  left: 50%;
  max-width: 100%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export function Splash({...rest}: SplashProps) {
    return <Root {...rest}>
        <SplashCopy>
            <Heading>Alasdair Macrae</Heading>
            <Paragraph align="center" fontSize="large"><em>Web Developer and Trampolinist</em></Paragraph>
        </SplashCopy>
    </Root>
}