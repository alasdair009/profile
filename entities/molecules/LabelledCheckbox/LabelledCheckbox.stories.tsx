import { Meta, StoryObj } from "@storybook/nextjs";
import { LabelledCheckbox } from "./LabelledCheckbox";
import { expect, userEvent, waitFor, within } from "storybook/test";
import { BaseCheckbox, colors } from "@/entities";

const meta: Meta<typeof LabelledCheckbox> = {
  component: LabelledCheckbox,
  args: {
    name: "checkboxentry",
    required: true,
    isInvalid: false,
    label: "Checkbox entry",
  },
};
export default meta;

export const Default: StoryObj<typeof LabelledCheckbox> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const checkboxElement = canvas.getByTestId(BaseCheckbox.name);
    const labelElement = canvas.getByText("Checkbox entry");

    await expect(checkboxElement).toBeInTheDocument();
    await expect(labelElement).toBeInTheDocument();

    await userEvent.click(checkboxElement);

    await waitFor(() => expect(checkboxElement).toHaveProperty("checked"));
  },
};

export const Invalid: StoryObj<typeof LabelledCheckbox> = {
  args: {
    isInvalid: true,
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const checkboxElement = canvas.getByTestId(BaseCheckbox.name);

    await expect(checkboxElement).toHaveStyle({
      borderColor: colors.redHeat,
    });
  },
};
