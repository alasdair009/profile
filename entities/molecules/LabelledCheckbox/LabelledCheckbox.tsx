import { Root } from "./styles";
import { BaseCheckbox, BaseLabel } from "@/entities";
import { BaseCheckboxProps } from "@/entities/atoms/BaseCheckbox/BaseCheckbox";

type LabelledCheckboxProps = {
  /**
   * Text for the label.
   */
  label: string;
} & BaseCheckboxProps;

/**
 * Labelled checkbox.
 */
export function LabelledCheckbox({
  isInvalid,
  name,
  label,
  ...rest
}: LabelledCheckboxProps) {
  return (
    <Root data-testid={LabelledCheckbox.name}>
      <BaseLabel as="span">{label}</BaseLabel>
      <BaseCheckbox isInvalid={isInvalid} {...rest} />
    </Root>
  );
}
