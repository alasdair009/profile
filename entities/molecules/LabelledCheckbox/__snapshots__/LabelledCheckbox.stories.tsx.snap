// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/LabelledCheckbox Default play-test 1`] = `
<div data-testid="LabelledCheckbox"
     class="sc-braxZu ldDAHa"
>
  <span data-testid="BaseLabel"
        class="sc-dsLQwm likJvJ"
  >
    Checkbox entry
  </span>
  <input data-testid="BaseCheckbox"
         required
         class="sc-elDIKY HOnpQ"
         type="checkbox"
  >
</div>
`;

exports[`molecules/LabelledCheckbox Invalid play-test 1`] = `
<div data-testid="LabelledCheckbox"
     class="sc-braxZu ldDAHa"
>
  <span data-testid="BaseLabel"
        class="sc-dsLQwm likJvJ"
  >
    Checkbox entry
  </span>
  <input data-testid="BaseCheckbox"
         required
         class="sc-elDIKY dKDQel"
         type="checkbox"
  >
</div>
`;
